### ISSUE TITLE
Firebase Database Paging Support - [New Feature-Pagination]

### ISSUE THREAD

**Issue by PatilShreyas** [firebase/FirebaseUI-Android#1601]
Firebase Realtime Database has **FirebaseRecyclerAdapter** to show Data Items in RecyclerView. But, the issue is it has not Paging support added until now. I have developed a solution on Pagination of Firebase Database in RecyclerView on below repository with library code and sample app.
https://github.com/PatilShreyas/FirebaseRecyclerPagination

I referred source code of FirestorePagingAdapter while developing the solution.
FirebaseRecyclerPagingAdapter is built on top of Android Paging Support Library.
So, I request you to please permit me to add this feature in the official FirebaseUI library to help developers with this paging library to load data in pages.
Can I fork this repository and make this feature available?

**Comment by samtstern** [comment]
@PatilShreyas thanks for filing this!  So since you based this on the Firestore adapter, does that mean the tradeoffs are the same?  In particular, does the adapter ignore realtime updates?

**Comment by PatilShreyas** [comment]
Yes. Till now, it does not support realtime updates.

**Comment by PatilShreyas** [comment]
@samtstern Thank you !
I will do it.

**Comment by PatilShreyas** [comment]
@samtstern I have updated code as you suggested here #1603 
Implemented the same as Firestore version.

**Comment by PatilShreyas** [comment]
Hi @samtstern 
I go through [FirestorePagingAdapter](https://github.com/firebase/FirebaseUI-Android/blob/master/firestore/src/main/java/com/firebase/ui/firestore/paging/FirestorePagingAdapter.java) and there is a method in it.

```java
public void retry() {
        FirestoreDataSource source = mDataSource.getValue();
        if (source == null) {
            Log.w(TAG, "Called retry() when FirestoreDataSource is null!");
            return;
        }

        source.retry();
    }
```
I tried to implement an exact feature in our `FirebaseRecyclerPagingAdapter`.
But, I observed that `source` always remains null. Every time it shows Log...
_"Called retry() when FirestoreDataSource is null!"_. 
I also tested calling this method from [FirestorePagingActivity](https://github.com/firebase/FirebaseUI-Android/blob/master/app/src/main/java/com/firebase/uidemo/database/firestore/FirestorePagingActivity.java) after caught in error. Still, it failed.

How do we implement `retry()` in `FirebaseRecyclerPagingAdapter`

**Comment by samtstern** [comment]
This has been fixed and released in `5.0.0`

**Comment by SerggioC** [comment]
Does the adapter support realtime updates now? 
I need an implementation of paging with realtime updates of the current page if the database is updated in that interval of data.
